### PR TITLE
nfs-ganesha: added cleanup and unmount tasks

### PR DESCRIPTION
**Pull request by gauravsitlani**
Signed-off-by: gauravsitlani <gsitlani@redhat.com>

Successful [cleanup_test* logs](http://magna002.ceph.redhat.com/ceph-qe-logs/gaurav/ganesha-tests/ceph-qe-scripts-logs/)

### CODE REVIEW

File: rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-rgw_hostname:
-ganesha_config_exists: true # set true to create ganesha config based on the above input or set false not to create any config.
-already_mounted: # set true if nfs_ganesha is already mounted or false if not.
-nfs_version:
-nfs_mnt_point: # provide full path
-Pseudo:
+---
+user_id:  # Provide the User_Id from /etc/ganesha/ganesha.conf
+access_key:  # Provide the Access_Key_Id value from /etc/ganesha/ganesha.conf for the above User_Id
+secret_key:  # Provide the Secret_Access_Key value from /etc/ganesha/ganesha.conf for the above User_Id
```
Comment (rakeshgm) [inline rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml:4]: the information about the `access key` and `secret key` will be present in ganesha.conf only if ganesha.conf exists.  there is an option below to say that gaensha.conf does not exist. if ganesha.conf does not exist we can take it from radosgw-admin cli if we know about the user.  cab you correct the text here. ?

File: rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+rgw_hostname:  # Provide the hostname of the rgw node
+ganesha_config_exists: true  # set true to create ganesha config based on the above input or set false not to create any config.
+already_mounted:  # set true if nfs_ganesha is already mounted or false if not.
+cleanup:  # By default set to true to perform a cleanup after a test is executed successfully.
+do_unmount:  # By default set to true to perform an unmount on the ganesha mount_point after a test is executed successfully.
+nfs_version:  # Provide the nfs protocol version being used
```
Comment (rakeshgm) [inline rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml:10]: mention in the comment, to user `4` or `3` and not `4.1` or `3.1` etc.

File: rgw/v2/tests/nfs_ganesha/initialize.py [hunk 3]
```diff
@@ -92,7 +94,7 @@ def do_un_mount(self):
         log.info("un_mounting dir: %s" % self.rgw_user_info["nfs_mnt_point"])
         un_mount_cmd = "sudo umount %s" % self.rgw_user_info["nfs_mnt_point"]
         un_mounted = utils.exec_shell_cmd(un_mount_cmd)
-        if un_mounted:
+        if un_mounted != "":
```
Comment (rakeshgm) [inline rgw/v2/tests/nfs_ganesha/initialize.py:97]: add TODO comment here also which you have added [here](https://github.com/red-hat-storage/ceph-qe-scripts/pull/240/files#diff-d13c2a01592688685d9f1a566d9c7437750ef3ca59d00224cfc40feaca805519R166-R167)

Comment (gauravsitlani) [reply rgw/v2/tests/nfs_ganesha/initialize.py:99]: sure added

File: rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml (hunk 1 above)
Comment (gauravsitlani) [reply rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml]: sure updated the text over there.

File: rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml (hunk 2 above)
Comment (gauravsitlani) [reply rgw/v2/tests/nfs_ganesha/config/rgw_user.yaml]: Added the changes, mentioning Accepted values : 3 or 4

**Review by rakeshgm — APPROVED** [review]

**Pull request merged**